src/appleseed/renderer/modeling/light/sunlight.cpp

```diff
@@ -141,17 +148,24 @@ namespace
         virtual void sample(
             const ShadingContext&   shading_context,
             const Transformd&       light_transform,
+            const Vector3d&         target_point,
             const Vector2d&         s,
             Vector3d&               position,
             Vector3d&               outgoing,
             Spectrum&               value,
             float&                  probability) const override
         {
+            //angular_diameter = one_radian * sun_diameter / distance
+            //sun_diameter = angular_diameter * distance / one_radian = 0.5331 degrees * 1.496 * 10^8 km / 57.29578 degrees
+            double sun_diameter = 0.5331 * m_safe_scene_diameter / 57.29578;
+            sun_diameter *= m_values.m_size_multiplier;
+
             sample_disk(
                 light_transform,
+                target_point,
                 s,
                 m_scene_center,
-                m_scene_radius,
+                sun_diameter/2,
```

Comment:
Would make sense to compute the Sun's _radius_ directly, thus avoiding a division.